src/dlgTriggerEditor.cpp

```diff
@@ -6957,25 +6957,28 @@ void dlgTriggerEditor::slot_script_main_area_delete_handler()
 
 void dlgTriggerEditor::slot_script_main_area_add_handler()
 {
+    auto createItemList = [](auto * mpScriptsMainArea) {
+        auto pItem = new QListWidgetItem;
+        pItem->setText(mpScriptsMainArea->lineEdit_script_event_handler_entry->text());
+        mpScriptsMainArea->listWidget_script_registered_event_handlers->addItem(pItem);
+        return pItem;
+    };
+
     mpScriptsMainArea->trimEventHandlerName();
     if (mIsScriptsMainAreaEditHandler) {
         if (!mpScriptsMainAreaEditHandlerItem) {
             mIsScriptsMainAreaEditHandler = false;
-            goto LAZY;
+            auto pItem = createItemList(mpScriptsMainArea);
```

Comment:
Using a lambda function does seem a rather elegant solution to wanting the same block of code in ~two~, no *three* places.  Looking more closely I see the return value is never used - so is it necessary? Lambdas are not something I fully grok yet - so I will have to defer to someone else who has a better understanding of this area. *\* yells: @vadi2 \**

Reply:
i improved the lambda function just now(then recent..) mind recheck it?